lib/addons/src/index.js

```diff
@@ -52,7 +53,16 @@ export class AddonStore {
   }
 
   addPanel(name, panel) {
-    this.panels[name] = panel;
+    // supporting legacy addons, which have not migrated to the active-prop
+    const original = panel.render;
+    if (original && original.toString() && !original.toString().match(/active/)) {
+      this.panels[name] = {
+        ...panel,
+        render: ({ active }) => TabWrapper({ active, render: original }),
```

Comment:
I feel pretty confused about what is going on here. Isn't this throwing away most of the implementation of the original panel? In the docs we talk about `componentDidX` etc -- creating a new component that just contains the `render` function isn't going to work in general.

Reply:
This works just fine, TabWrapper is called when render is called. TabWrapper calls the render prop.

```js

Reply:
Yeah but about the other lifecycle methods on `panel`?

Reply:
I can see that all the addons are implemented with something like `active => active ? <X> : null` (even in the docs) . What are the other ways of using the `active` field (What am I missing) ?
I just don't understand why we need to controll this. Why just not to use `TabWrapper` always and hide the "activation" from the users?

Reply:
@igor-dv The usecase is: